eZ/Publish/Core/Base/Container/Compiler/FieldTypeCollectionPass.php

```diff
@@ -63,14 +65,13 @@ public function process(ContainerBuilder $container)
                 $fieldTypeCollectionFactoryDef->addMethodCall(
                     'registerFieldType',
                     [
-                        // Only pass the service Id since field types will be lazy loaded via the service container
-                        $id,
                         $attribute['alias'],
+                        new Reference($id),
                     ]
                 );
 
                 // Add FieldType to the "concrete" list if it's not a fake.
-                if (!is_a($container->findDefinition($id)->getClass(), '\eZ\Publish\Core\FieldType\Null\Type', true)) {
+                if (!is_a($container->findDefinition($id)->getClass(), Type::class, true)) {
```

Comment:
```suggestion
                if (!is_a($container->findDefinition($id)->getClass(), NullType::class, true)) {
```

According to documentation `is_a` function is deprecated since PHP 5.0 so maybe we can change it to:

A) explicitly use `Reflection API` 
B) `$container->findDefinition($id)->getClass() === NullType:class` and ignore subclasses (BC break?)

Reply:
As of PHP 5.3, `is_a` is no longer deprecated: https://www.php.net/manual/en/function.is-a.php